### ISSUE TITLE
Import information from text (or other simple) file

### ISSUE THREAD

**Issue by BenBergman** [partkeepr/PartKeepr#163]
While installing PartKeepr, I needed to insert a large number of categories and sub categories into the database. I am not very familiar with SQL, so it is possible I could have done something directly through that, but it would be nice if there was some way to batch insert information into PartKeepr. Our categories were in a plain text file, but could have been in some form of spreadsheet, cvs, etc. file. Perhaps you could have an option to import a file into a category, with one sub-category per line.

In addition to this, it might be nice to export data into some easy to use (by humans) format like a csv for improved data portability.

Just a thought for a future improvement. I know there are many more important issues to cover first.
##

**Comment by varalu** [comment]
+1 for this feature.

**Comment by felicitus** [comment]
The feature is mostly completed; however, the user must be asked what kind of separator a CSV file uses.

**Comment by brads4** [comment]
CSV stands for Comma Separated Value. Would be best to default to this, and allow the user to override, rather than have them input what will become the same data every single time they import. 

Less clicks == happy users.

**Comment by felicitus** [comment]
@brads4 yes and no. Whilst CSV means comma separated value, quite many software packages use a semicolon rather than a comma for separation. [https://www.ietf.org/rfc/rfc4180.txt RFC4180] defines the comma as mandatory separator. Both KiCad and Eagle, for example, use a semicolon to separate fields. AFAIR a semicolon is also the default with Excel, but I can't verify that at the moment. LibreOffice asks for the format upon export, Google Spreadsheets use a comma.

**Comment by brads4** [comment]
I am aware. My comment was related to your user input design decision, where you mentioned forcing the user to type in a separator character every time they import.  Have the field filled in with a default value of a comma, and allow the user to override it.  This ensures that the most common case scenario (folks who use commas) don't have to type in a value every time.  Better user experience is the only point I was addressing, as I am aware lots of folks use Delimiter Separated Values (DSV) as CSV, which I think is reasonable to support in the way you have.
